PxtlCa.XmlCommentMarkDownGenerator.MSBuild/GenerateMarkdown.cs

```diff
@@ -0,0 +1,273 @@
+        {
+            var lines = File.ReadLines(filePath);
+            var firstDashedLine = lines.FirstOrDefault() ?? string.Empty;
+            if (firstDashedLine.StartsWith("---"))
+            {
+                var followingLines = lines.Skip(1).TakeWhile(line => !line.StartsWith("---")).ToList();
+                if(followingLines.Count == 0)
+                {
+                    frontMatter = firstDashedLine;
+                    isEmpty = true;
+                    return true;
+                }
+                else
+                {
+                    followingLines.Insert(0, firstDashedLine);
+                    frontMatter = String.Join(Environment.NewLine, followingLines);
+                    isEmpty = false;
+                    return true;
+                }
+            }
+            frontMatter = string.Empty;
+            isEmpty = true;
+            return false;
+        }
+
+        private void ReadOptionsFromString(string frontMatter)
+        {
+            var input = new StringReader(frontMatter);
+            var deserializer = new DeserializerBuilder()
+                .WithNamingConvention(new CamelCaseNamingConvention())
+                .Build();
+            var options = deserializer.Deserialize<YamlOptions>(input);
+            MergeFiles = options.MergeXmlComments;
+            if(Enum.TryParse<AllowedTagOptions>(options.AllowedCustomTags, true, 
+                    out AllowedTagOptions result))
+            {
+                //warn (rather than treat as error condition)
+                //if user indicates one of the two currently used options
+                WarnOnUnexpectedTag = result == AllowedTagOptions.All;
+            }
+        }
+
+        /// <summary>
+        /// for testing.  
+        /// sets the exception for throw outside the catch
+        /// </summary>
+        public Exception LoggedException { get; set; }
+
+        private void Merge()
+        {
+            //get all md files in Documentation Path
+            //except those generated in this task
+            var otherMDFiles = Directory.EnumerateFiles(DocumentationPath.ItemSpec, "*.md", SearchOption.AllDirectories).ToList();
+            otherMDFiles = otherMDFiles.Except(GeneratedMDFiles).ToList();
+            var mergeInto = otherMDFiles.FirstOrDefault();
+            if (null == mergeInto)
+            {
+                mergeInto = GeneratedMDFiles.First();
+                File.Copy(mergeInto, OutputFile.ItemSpec, true);
+                foreach (var mdFile in GeneratedMDFiles.Skip(1))
+                {
+                    File.AppendAllText(OutputFile.ItemSpec, Environment.NewLine);
+                    File.AppendAllText(OutputFile.ItemSpec, File.ReadAllText(mdFile));
+                }
+            }
+            else
+            {
+                File.Copy(mergeInto, OutputFile.ItemSpec, true);
+                foreach (var mdFile in otherMDFiles.Skip(1))
```

Comment:
Why are we skipping the first otherMDFile?  I'm not understanding that.  If it's about the YAML string, can we just open it up and skip the YAML?

The rest of this makes sense - first we copy all the OtherMDFiles (except the first) into the target file, and then the generated files.  That makes sense.  OtherMDFiles become the header of the thing